spyder/widgets/editor.py

```diff
@@ -1302,7 +1304,11 @@ def file_saved_in_other_editorstack(self, index, filename):
     def select_savename(self, original_filename):
         self.redirect_stdio.emit(False)
         filename, _selfilter = getsavefilename(self, _("Save file"),
-                                               original_filename)
+                                               original_filename,
+                                               get_edit_filters(),
+                                               get_filter(get_edit_filetypes(),
+                                                          osp.splitext(original_filename)[1])
+                                               )

```

Comment:
Hi @goanpeca these are my results just look like that on:

* Win 7 
![fix-save_as win 7 seleccion_006](https://cloud.githubusercontent.com/assets/5307894/19621902/1925cb18-9862-11e6-9992-b17b560a29df.png)

* Linux
![fix-save_as seleccion_006](https://cloud.githubusercontent.com/assets/5307894/19621906/2ed1257a-9862-11e6-93a3-898843e9a4e9.png)